src/identity/apis/auth.ts

```diff
@@ -0,0 +1,213 @@
+  const response = await getIdentity({})
+
+  if (response.status === 401) {
+    throw new UnauthorizedError(response.data.message)
+  }
+
+  if (response.status === 500) {
+    throw new ServerError(response.data.message)
+  }
+
+  const user = response.data
+  return user
+}
+
+// fetch user identity from /quartz/me.
+export const fetchQuartzMe = async (): Promise<MeQuartz> => {
+  const response = await getMeQuartz({})
+
+  if (response.status === 401) {
+    throw new UnauthorizedError(response.data.message)
+  }
+
+  if (response.status === 404) {
+    throw new NotFoundError(response.data.message)
+  }
+
+  if (response.status === 500) {
+    throw new ServerError(response.data.message)
+  }
+
+  const user = response.data
+  return user
+}
+
+// fetch user identity from /me (used in OSS and environments without Quartz)
+const fetchLegacyIdentity = async (): Promise<UserResponseIdpe> => {
+  const response = await getMeIdpe({})
+
+  if (response.status === 500) {
+    const error: IdpeError = response.data
+    throw new ServerError(error.message)
+  }
+
+  const user = response.data
+  return user
+}
+
+// fetch details about user's current account
+export const fetchAccountDetails = async (
+  accountId: string | number
+): Promise<Account> => {
+  const accountIdString = accountId.toString()
+
+  const response = await getAccount({
+    accountId: accountIdString,
+  })
+
+  if (response.status === 401) {
+    throw new UnauthorizedError(response.data.message)
+  }
+
+  if (response.status === 500) {
+    throw new ServerError(response.data.message)
+  }
+
+  const accountDetails = response.data
+  return accountDetails
+}
+
+// fetch details about user's current organization
+export const fetchOrgDetails = async (orgId: string): Promise<Organization> => {
+  const response = await getOrg({orgId})
+
+  if (response.status === 401) {
+    throw new UnauthorizedError(response.data.message)
+  }
+
+  if (response.status === 500) {
+    throw new ServerError(response.data.message)
+  }
+
+  const orgDetails = response.data
+  return orgDetails
+}
+
+// Transform a response from `/identity` into a response that Me machinery can use
+export const convertIdentityToMe = (
+  quartzIdentity: CurrentIdentity
+): MeQuartz => {
+  const {account, org, user} = quartzIdentity
+
+  return {
+    // User Data
+    email: user.email,
+    id: user.id,
+    // Careful about this line.
+    isOperator: user.operatorRole ? true : false,
+    operatorRole: user.operatorRole,
+
+    // Account Data
+    accountCreatedAt: account.accountCreatedAt,
+    accountType: account.type,
+    paygCreditStartDate: account.paygCreditStartDate,
+    billingProvider: account.billingProvider ? account.billingProvider : null,
+
+    // Organization Data
+    clusterHost: org.clusterHost,
+    regionCode: org.regionCode ? org.regionCode : null,
+    isRegionBeta: org.isRegionBeta ? org.isRegionBeta : null,
+    regionName: org.regionName ? org.regionName : null,
+  }
+}
```

Comment:
make sure you get the latest master - this file shouldn't be in this diff, since it's already a part of master

Reply:
Gotcha - I think that's from me pulling the file into my feature branch before your PR made it to `master`.